src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorProjectEngine.cs

```diff
@@ -122,6 +123,11 @@ protected override RazorCodeDocument CreateCodeDocumentCore(RazorProjectItem pro
                 codeDocument.SetFileKind(fileKind);
             }
 
+            if (cssScope != null)
```

Comment:
Not sure if you want to do the work now or not but the "design time" code path provides all of the inputs upfront:
```C#
public override RazorCodeDocument ProcessDesignTime(RazorSourceDocument source, string fileKind, IReadOnlyList<RazorSourceDocument> importSources, IReadOnlyList<TagHelperDescriptor> tagHelpers)
```

If you want the design time experience to understand that those CSS scopes are on the attributes you'll need to update the `ProcessDesignTime` signature and the corresponding consumption sites in aspnetcore-tooling. That being said, there's no reason that can't be done later so i'll leave it up to you.

Reply:
There's no design-time equivalent to the CSS scope attributes. It's a no-op at design time.